addons/crm_iap_lead/models/crm_iap_lead_mining_request.py

```diff
@@ -190,12 +191,16 @@ def _perform_request(self):
         }
         try:
             response = iap_tools.iap_jsonrpc(endpoint, params=params, timeout=300)
+            if response.get('data'):
+                return response['data']
+            else:
+                self.error_code = 'no_result'
+                return False
             return response['data']
         except iap_tools.InsufficientCreditError as e:
-            self.error = 'Insufficient credits. Recharge your account and retry.'
+            self.error_code = 'insufficient_credits'
```

Comment:
Shouldn't we catch "other execptions" in a generic/unknown error code ?

Reply:
In case of another error, the code will raise the exception, which will result in a standard error display for the end user (same as a UserError I'm guessing, you get a modal with the exception text).

We could store a "generic" error_type, but then I would probably need to keep the "error" field to store the Exception message.
I don't feel like it's necessary to keep so many fields.

Reply:
"I don't feel like it's necessary to keep so many fields." Fields ? Keys maybe ?

Reply:
To explain a bit more.

When the error is "handled", one of insufficient_credits or no_results, we show it in the form view with a small banner message like so:

![Selection_332](https://user-images.githubusercontent.com/41282536/118784611-743eb480-b890-11eb-9ad3-8fa68a99da2b.png)

When the error is not handled (generic Exception), I changed the code a bit so it's displayed like this:

![Selection_331](https://user-images.githubusercontent.com/41282536/118784726-89b3de80-b890-11eb-861f-5fce30a61b10.png)

No I technically _could_ show a generic error in the form view as well with the same red banner message.
But either I just show "There was a generic error.", and only use an additional error_type key, either I show "There was a generic error: Prout", but then that means I need to keep the previously existing error field to store "Prout", and my new field "error_type" to store "generic".

Since we don't really care about generic errors, I though raising them as a UserError was sufficient.

Tell me what you think.

Side note: I hope you like me example :D

Reply:
That's really enlightening. Examples are :+1:

Reply:
Ok then :D